app/invocation/invocation_action_card.tsx

```diff
@@ -0,0 +1,70 @@
+import pako from "pako";
+import React from "react";
+import format from "../format/format";
+import InvocationModel from "./invocation_model";
+import { build } from "../../proto/remote_execution_ts_proto";
+import rpcService from "../service/rpc_service";
+
+interface Props {
+  model: InvocationModel;
+  inProgress: boolean;
+  search: string;
+}
+
+interface State {
+  loading: boolean;
+  contents: ArrayBuffer;
+  action_array: Uint8Array;
+  action_obj: build.bazel.remote.execution.v2.Action;
+  error: string;
+}
+
+export default class ActionCardComponent extends React.Component {
+  props: Props;
+
+  state: State = {
+    loading: false,
+    contents: new ArrayBuffer(8),
+    action_array: new Uint8Array(),
+    action_obj: null,
+    error: "",
+  };
+
+  fetchAction() {
+    let actionFile = "bytestream://localhost:1987/blobs/" + this.props.search.substring(1);
+    console.log(actionFile);
+    rpcService
+      .fetchBytestreamFile(actionFile, this.props.model.getId(), "arraybuffer")
+      .then((action_buff: any) => {
+        let temp_array = Uint8Array.from(action_buff);
+        this.setState({
+          ...this.state,
+          contents: action_buff,
+          action_array: temp_array,
+          action_obj: build.bazel.remote.execution.v2.Action.decode(temp_array),
+          loading: true,
```

Comment:
The pattern we usually follow for our other components is to set `loading: true` just before starting an RPC, and then set `loading: false` after the RPC completes.

I usually put it in a `finally` handler so that it gets set to false both on success and error).

So like

```
this.setState({loading: true});
rpcService.fetch(...).then(...).catch(...).finally(() => this.setState({loading: false})
```